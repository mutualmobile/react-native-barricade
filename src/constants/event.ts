export const EventType = {
  ShowBarricadeView: 'ShowBarricadeView',
};
