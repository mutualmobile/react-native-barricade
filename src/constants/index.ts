export * from './event';
export * from './strings';
export * from './unicode';
