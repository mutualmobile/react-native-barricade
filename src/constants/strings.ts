export const Strings = {
  Back: 'Back',
  Barricade: 'Barricade',
  DisableBarricade: 'Disable Barricade',
  Done: 'Done',
  EnableBarricade: 'Enable Barricade',
  Reset: 'Reset',
  Space: ' ',
};
