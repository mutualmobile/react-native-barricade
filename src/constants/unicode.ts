export const Unicode = {
  ChevronLeft: '\u2039',
  CheckMark: '\u2713',
  ChevronRight: '\u203A',
};
