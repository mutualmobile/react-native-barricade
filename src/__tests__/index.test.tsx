/**
 * @jest-environment jsdom
 */
import React from 'react';
import { DeviceEventEmitter, DevSettings } from 'react-native';
import renderer from 'react-test-renderer';
import { EventType, Strings } from '../constants';

describe('given that its not __DEV__ mode', () => {
  let Barricade;
  beforeAll(() => {
    (global as any).__DEV__ = false;
    Barricade = require('../index');
  });

  describe('and barricade is undefined,', () => {
    test('should match snapshot', () => {
      const tree = renderer.create(<Barricade.BarricadeView />);

      expect(tree).toMatchSnapshot();
      expect.assertions(1);
    });

    test('getBarricadeInstance should return undefined', () => {
      expect(Barricade.getBarricadeInstance()).toBe(undefined);
      expect.assertions(1);
    });
  });

  test('and createBarricade is called, should create barricade instance and add Barricade to debug menu', () => {
    const spy = jest.spyOn(DevSettings, 'addMenuItem');

    const barricade = Barricade.createBarricade();

    expect(barricade).not.toBe(undefined);
    expect(spy).not.toHaveBeenCalledWith(
      Strings.Barricade,
      Barricade.showBarricadeView,
    );
    expect.assertions(2);
  });

  test('and showBarricadeView is called, should not emit ShowBarricadeView event', () => {
    const spy = jest.spyOn(DeviceEventEmitter, 'emit');

    Barricade.showBarricadeView();

    expect(spy).not.toHaveBeenCalledWith(EventType.ShowBarricadeView);
    expect.assertions(1);
  });

  describe('and barricade instance is created,', () => {
    let BarricadeInstance;
    beforeAll(() => {
      BarricadeInstance = Barricade.createBarricade();
    });

    test('getBarricadeInstance should return Barricade instance', () => {
      expect(Barricade.getBarricadeInstance()).toBe(BarricadeInstance);
      expect.assertions(1);
    });
  });
});
