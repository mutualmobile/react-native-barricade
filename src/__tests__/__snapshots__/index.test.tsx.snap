// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given that its not __DEV__ mode and barricade is undefined, should match snapshot 1`] = `null`;
